planner/languages/csharp/csharp_planner.go

```diff
@@ -17,9 +29,88 @@ func (p *Planner) Name() string {
 }
 
 func (p *Planner) IsRelevant(srcDir string) bool {
-	return false
+	a, err := plansdk.NewAnalyzer(srcDir)
+	if err != nil {
+		// We should log that an error has occurred.
+		return false
+	}
+	return a.HasAnyFile("*.csproj")
 }
 
 func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
-	return &plansdk.Plan{}
+	plan, err := p.getPlan(srcDir)
+	if err != nil {
+		// Added this Printf because `devbox shell` was silently swallowing this error.
+		// TODO savil. Have `devbox shell` error out or print it instead.
+		fmt.Printf("error in getPlan: %s\n", err)
```

Comment:
The issue is this: imagine there's 2 planners, one of them errors, one of them succeeds – `devbox shell` succeeds because it's able to use the successful plan – but this would still print out an error. 

I'm wondering if instead we should log an error after we've run all the planners and are trying to pick what plan to use.

Question: in practice, what type of error where you seeing here?

Reply:
This happened when I was trying out dotnet-6 but hadn't added support for it in the code below. 

I expected `devbox shell` to error with "unsupported version", but it just continued on -- which I suppose is the designed for behavior. But maybe we need to revisit that in some form: once a Planner has been deemed relevant, then it should not error.